core/src/test/java/io/questdb/griffin/SqlParserTest.java

```diff
@@ -257,7 +257,12 @@ public void testAsOfJoinSubQuery() throws Exception {
     public void testAsOfJoinSubQueryInnerPredicates() throws Exception {
         // which means "where" clause for "e" table has to be explicitly as post-join-where
         assertQuery(
-                "select-choose c.customerId customerId, e.blah blah, e.lastName lastName, e.employeeId employeeId, e.timestamp timestamp, o.customerId customerId1 from (select [customerId] from customers c asof join select [blah, lastName, employeeId, timestamp] from (select-virtual ['1' blah, lastName, employeeId, timestamp] '1' blah, lastName, employeeId, timestamp from (select [lastName, employeeId, timestamp] from employees where lastName = 'x') where blah = 'y' order by lastName) e on e.employeeId = c.customerId join select [customerId] from orders o on o.customerId = c.customerId) c",
+                "select-choose c.customerId customerId, e.blah blah, e.lastName lastName, e.employeeId employeeId, e.timestamp timestamp, o.customerId customerId1 from " +
+                        "(select [customerId] from customers c " +
+                        "asof join select [blah, lastName, employeeId, timestamp] from " +
+                        "(select-virtual ['1' blah, lastName, employeeId, timestamp] '1' blah, lastName, employeeId, timestamp from " +
+                        "(select [lastName, employeeId, timestamp] from employees) " +
+                        "order by lastName) e on e.employeeId = c.customerId post-join-where e.lastName = 'x' and e.blah = 'y' join select [customerId] from orders o on o.customerId = c.customerId) c",
```

Comment:
this looks like a new bug to me `e.lastName = 'x' and e.blah = 'y'` should not be executed post-join, those are pre-join filters

Reply:
These single table conditions are pushed to post join phase of ASOF join . 
While changing left join condition and pushing down to right table is generally unsafe .
Let me double check my notes .

Reply:
Right, pushing down predicates into right table of left join can easily distort join output .
Same as above - it would only work if we both copied it right model and kept as post join condition .